akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala

```diff
@@ -207,6 +214,21 @@ class Http2ServerSpec extends AkkaSpec with WithInPendingUntilFixed with Eventua
           remainingToServerWindowFor(TheStreamId) shouldBe 0
           expectNoWindowUpdates(100.millis) // might fail here until all buffers have been filled
         }
+      "fail if advertised content-length doesn't match (actual entity is longer)" in new WaitingForRequest(
```

Comment:
I think this also needs to check that `RST_STREAM` was sent for the stream and that later streams still work.